code/game/objects/items/weapons/material_trash.dm

```diff
@@ -0,0 +1,72 @@
+/obj/item/weapon/material_trash
+	name = "trash"
+	icon = 'icons/obj/material_trash.dmi'
+	var/matter_chances = list()	//List of lists: list(material_name, chance, amount)
+
+
+/obj/item/weapon/material_trash/New()
+	. = ..()
```

Comment:
Use `Initialize` for those. 

`. = ..()` should be the first line in all `Initialize` procs.